### PR TITLE
Set pockets to autopickup and unload

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Features "You can disable specific pockets on items and disable unloading specific pockets"

#### Purpose of change
Been playing with the additional pockets that I made in #53162. It would be nice if these pockets didn't unload when I unload my bag. My proper bag is full of loot but it would be nice if when I dump 60 things the AR and Haligan strapped to the back stay where they are instead of getting unloaded too. 

#### Describe the solution
In the autopickup favorites menu there is now an option to disable auto pickup per pocket and disable unload per pocket:

- Disable auto pickup works the same as whitelisting something that didn't fit in the pocket did before (which was the workaround I used for this functionality). That is to say items will no longer go into the pocket automatically but still will if you manually call "insert" on the item. However how this now differs is since this new features sole purpose is disabling autopickups it has the added benefit of pockets which are disabled only contributing there "contained volume" to your available volume continuing my push from #53079 to make the volume UI clearer. 
- Disable unload just means when you do either the unload into tile or unload actions any pockets with unload disabled won't unload with the rest of your items.
- When using unload into tile, favorite items will not be unloaded (this is not the case for traditional unload I have an explanation for that in the comments) 

#### Describe alternatives you've considered
Manually removing all items from my backpack after each loot run,
Wearing two backpacks.

#### Testing
New favorites UI:
image

Inventory before an unload (with that backpack):
image

Inventory after an unload:
image

Messenger bag set to disabled (note the volume is exactly the contents):
image

#### Additional context
The basic unload function didn't specify containers before. In my testing and also code hunting changing this doesn't break anything but someone with more code experience should at least mull this over.

### CODE REVIEW

File: src/item_contents.cpp [hunk 1]
```diff
@@ -1124,11 +1134,18 @@ std::list<item *> item_contents::all_items_top( const std::function<bool( item_p
...
-    } );
+    if( unloading ) {
+        return all_items_top( [pk_type]( item_pocket & pocket ) {
+            return pocket.is_type( pk_type ) && pocket.settings.is_unloadable();
+        } );
+    } else {
+        return all_items_top( [pk_type]( item_pocket & pocket ) {
+            return pocket.is_type( pk_type );
+        } );
+    }
```
Comment (andrei8l) [inline src/item_contents.cpp:1148]:
```suggestion
    return all_items_top( [pk_type]( item_pocket & pocket ) {
          return pocket.is_type( pk_type ) && ( !unloading || pocket.settings.is_unloadable() );
    } );
```

File: src/item_contents.cpp [hunk 2]
```diff
@@ -132,6 +134,14 @@ bool pocket_favorite_callback::key( const input_context &, const input_event &ev
         popup.title( string_format( _( "Enter Priority (current priority %d)" ),
                                     selected_pocket->settings.priority() ) );
         selected_pocket->settings.set_priority( popup.query_int() );
+    } else if( input == 'd' ) {
+        //toggle the disabled setting for this pocket
+        selected_pocket->settings.set_disabled( !selected_pocket->settings.is_disabled() );
+        return true;
+    } else if( input == 'u' ) {
+        //toggle the unload setting for this pocket
```
Comment (andrei8l) [inline src/item_contents.cpp:142]: These comments are useless

**Review by andrei8l — COMMENTED** [review]
> Disable unload just means when you do either the unload into tile or unload actions any pockets with unload disabled won't unload with the rest of your items.

The multidrop UI ignores favorites when selecting all and it'd be nice to have the same functionality here for consistency instead of a new setting with a different behavior.

File: src/item_contents.cpp (hunk 1 above)
Comment (bombasticSlacks) [reply src/item_contents.cpp:1148]: This is much harder to read and understand and a fraction slower I don't really think it's worth changing.

**Comment by bombasticSlacks** [comment]
Changing what drops by default seems like more of a conversation than adding an additional opt in feature. 

As well its a separate bit of logic since it interacts with the items rather than the pockets themselves. I'm sure someone else could PR that.

File: src/item_contents.cpp (hunk 1 above)
Comment (andrei8l) [reply src/item_contents.cpp:1146]: Enough people disagreed that there's even a clang-tidy check for [this](https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html). This project hasn't formally adopted that style though so it's your call.

**Comment by Inglonias** [comment]
I feel that this is a little unnecessary. If I don't want things to drop from my inventory, I mark them as favorites and move on.

**Comment by bombasticSlacks** [comment]
and yet there is another way to interact with getting rid of things other than dropping them #52201

**Comment by bombasticSlacks** [comment]
I've added the no favorites functionality to unload into tile. At this point I've spent longer arguing about it than the code change would take. 

Favorites ignore shouldn't be added to the regular unload as it tries to put the items into other containers in your inventory its default behavior isn't to drop.

**Comment by I-am-Erk** [comment]
You might want to merge with master just to kick these pending tests

File: src/item_contents.cpp [hunk 3]
```diff
@@ -132,6 +134,12 @@ bool pocket_favorite_callback::key( const input_context &, const input_event &ev
         popup.title( string_format( _( "Enter Priority (current priority %d)" ),
                                     selected_pocket->settings.priority() ) );
         selected_pocket->settings.set_priority( popup.query_int() );
+    } else if( input == 'd' ) {
```
Comment (kevingranade) [inline src/item_contents.cpp:137]: Oh what the hell, why isn't this using contexts?
(Not your fault)

**Pull request merged**